GetStream/stream-chat-swift
# [CIS-2009] Sync mock server with the backend

### 🔗 Issue Links

- https://stream-io.atlassian.net/browse/CIS-2009

### 🎯 Goal

- To keep back-end HTTP and Websocket responses up-to-date in the Mock Server

### 📝 Summary

- Sync will be performed every 1st and 15th of the month at 0:00
    - Json templates will be collected via [AIOHTTP](https://github.com/aio-libs/aiohttp) and [Stream REST API](https://getstream.io/chat/docs/rest)
- PR with updated json templates will be opened from `ci/sync-mock-server-{timestamp}` to `develop`
    - E2E tests will be executed based on new templates

### 🎨 Showcase

- [Sync example](https://github.com/GetStream/stream-chat-swift/actions/runs/3594999477):

<img width="820" alt="Screen Shot 2022-12-01 at 4 06 54 PM" src="https://user-images.githubusercontent.com/20794991/205101964-b29c39f3-cae4-451a-89e9-41fe940360cb.png">

### 🧪 Manual Testing Notes

1. Install dependencies: `pip3 install -r requirements.txt`
2. Run `bundle exec fastlane sync_mock_server`
3. Check out local git changes

**PS**: tested both UIKit (on CI) and SwiftUI (locally) with auto-generated json templates

### ☑️ Contributor Checklist


### 🎁 Meme

![gif](https://media.giphy.com/media/TIM8PusKoHVZVckJds/giphy.gif)
